### DISCUSSION TITLE
Add an export option to insert @generated header

### DISCUSSION THREAD

**Discussion by 93578237** [tweag/nickel#1736] [Ideas]
https://generated.at/
Now I am doing this with sed like `sed -i '1 i\# This file is @generated by Nickel. Do not edit.' "$target"` after exporting to yaml/json format

**Comment by vkleen** [comment]
I think that would work fine for YAML, but JSON doesn't support comments by default. It probably depends on the particular parser implementation whether that works or not.

**Reply by 93578237** [reply]
Oh, yes. I use JSON with comments for GHA workflows/actions

**Reply by ad-si** [reply]
Could also just be a special field like:

```JSON
{
  "_⛔️_": "This file is generated by config.ncl. DO NOT EDIT IT MANUALLY!",
  …
}
```
